### PR TITLE
Logic: Add Job Type field and make Status editable

### PR DESCRIPTION
**Pull request by migfoo02**
Changes: 
- Added `Job Type` as an optional field in `Job`. 
- Changed implementation for `Status` to support `edit`. (might overlap with the previous commit here)

Implementation for `Job Type` includes:
- Support for `add` and `edit` commands
- Changed test cases
- Updated UI

### CODE REVIEW

File: src/main/java/seedu/application/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -199,6 +219,8 @@ public String toString() {
                 .add("company", company)
                 .add("role", role)
                 .add("deadline", deadline)
+                .add("status", status)
```
Comment (evanyan13) [inline src/main/java/seedu/application/logic/commands/EditCommand.java:250]: EditCommand seems to have been updated well to suit the need of editing jobType

File: src/main/java/seedu/application/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -88,6 +93,21 @@ public static Status parseStatus(String status) throws ParseException {
...
+     * Parses a {@code String JobType} into a {@code JobType}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code jobType} is invalid.
+     */
+    public static JobType parseJobType(String jobType) throws ParseException {
+        requireNonNull(jobType);
+        String trimmedJobType = jobType.trim();
+        if (!JobType.isValidJobType(trimmedJobType)) {
+            throw new ParseException(JobType.MESSAGE_CONSTRAINTS);
```
Comment (peiran18) [inline src/main/java/seedu/application/logic/parser/ParserUtil.java:106]: Would be good to add tests to cover this line

File: src/main/java/seedu/application/model/job/JobType.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.application.commons.util.AppUtil;
+
+/**
+ * Represents a Job's jobType in the application book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidJobType(String)}
+ */
+public class JobType {
```
Comment (peiran18) [inline src/main/java/seedu/application/model/job/JobType.java:11]: Consider creating a test for this class!

File: src/main/java/seedu/application/logic/commands/EditCommand.java [hunk 4]
```diff
@@ -30,21 +42,24 @@ public class EditCommand extends Command {
         + "[" + PREFIX_COMPANY + "COMPANY] "
         + "[" + PREFIX_DEADLINE + "DEADLINE] "
         + "[" + PREFIX_STATUS + "STATUS] "
+        + "[" + PREFIX_JOBTYPE + "JOB_TYPE] "
         + "[" + PREFIX_INDUSTRY + "INDUSTRY] "
         + "Example: " + COMMAND_WORD + " 1 "
         + PREFIX_ROLE + "Software Engineer "
-        + PREFIX_COMPANY + "Google "
```
Comment (czhiruo) [inline src/main/java/seedu/application/logic/commands/EditCommand.java:36]: should retain the whitespace after the string if not the message is going to be quite weird

File: src/main/java/seedu/application/model/job/JobType.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+    public static final String TO_ADD_JOB_TYPE = "TO_ADD_JOB_TYPE";
+    public static final JobType EMPTY_JOB_TYPE = new JobType(TO_ADD_JOB_TYPE);
+    public final String jobType;
+    /**
+     * Enum for job types
+     */
+    public enum JobTypes {
+        FULL_TIME, PART_TIME, INTERNSHIP, TEMPORARY, CONTRACT, FREELANCE, VOLUNTEER
+    }
```
Comment (peiran18) [inline src/main/java/seedu/application/model/job/JobType.java:25]: Love the variety of options!

File: src/main/java/seedu/application/logic/commands/EditCommand.java [hunk 6]
```diff
@@ -30,21 +42,24 @@ public class EditCommand extends Command {
...
-        + PREFIX_DEADLINE + "Dec 31 2023 1200 "
```
Comment (czhiruo) [inline src/main/java/seedu/application/logic/commands/EditCommand.java:37]: same as above

**Review by peiran18 — APPROVED** [review]

File: src/main/java/seedu/application/logic/parser/ParserUtil.java (hunk 2 above)
Comment (czhiruo) [inline src/main/java/seedu/application/logic/parser/ParserUtil.java:106]: could cover this line in PasrserUtilTest

**Review by czhiruo — APPROVED** [review]

**Pull request merged**